Tests/tests/test_FrontPanel.cpp

```diff
@@ -0,0 +1,523 @@
+    FrontPanelTextDisplayImplMock frontPanelTextDisplayImplIntMock;
+    FrontPanelConfigImplMock frontPanelConfigImplMock;
+    FrontPanelDsTest()
+        : FrontPanelTest()
+    {
+        device::FrontPanelIndicator::getInstance().impl = &frontPanelIndicatorImplMock;
+	device::FrontPanelIndicator::getInstance("Power").impl = &frontPanelIndicatorImplStringMock;
+        device::FrontPanelIndicator::getInstance(0).impl = &frontPanelIndicatorImplIntMock;
+	device::FrontPanelTextDisplay::getInstance("Text").impl = &frontPanelTextDisplayImplStringMock;
+	device::FrontPanelTextDisplay::getInstance(0).impl = &frontPanelTextDisplayImplIntMock;
+	device::FrontPanelConfig::getInstance().impl = &frontPanelConfigImplMock;
+        device::FrontPanelTextDisplay::getInstance().impl = &frontPanelTextDisplayImplMock;
+
+    }
+    virtual ~FrontPanelDsTest() override
+    {
+        device::FrontPanelIndicator::getInstance().impl = nullptr;
+	device::FrontPanelIndicator::getInstance("Power").impl = nullptr;
+	device::FrontPanelTextDisplay::getInstance("Text").impl = nullptr;
+	device::FrontPanelConfig::getInstance().impl = nullptr;
+        device::FrontPanelTextDisplay::getInstance().impl = nullptr;
+	device::FrontPanelTextDisplay::getInstance(0).impl = nullptr;
+        device::FrontPanelIndicator::getInstance(0).impl = nullptr;
+
+    }
+};
+
+class FrontPanelInitializedTest : public FrontPanelTest {
+protected:
+    IarmBusImplMock iarmBusImplMock;
+    IARM_EventHandler_t dsPanelEventHandler;
+    FrontPanelConfigImplMock frontPanelConfigImplMock;
+    FrontPanelIndicatorImplMock frontPanelIndicatorImplMock;
+    IARM_EventHandler_t dsFrontPanelModeChange;
+
+
+    FrontPanelInitializedTest()
+        : FrontPanelTest()
+    {
+        IarmBus::getInstance().impl = &iarmBusImplMock;
+        device::FrontPanelConfig::getInstance().impl = &frontPanelConfigImplMock;
+        device::FrontPanelIndicator::getInstance().impl = &frontPanelIndicatorImplMock;
+
+        ON_CALL(iarmBusImplMock, IARM_Bus_RegisterEventHandler(::testing::_, ::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [&](const char* ownerName, IARM_EventId_t eventId, IARM_EventHandler_t handler) {
+                    if ((string(IARM_BUS_PWRMGR_NAME) == string(ownerName)) && (eventId == IARM_BUS_PWRMGR_EVENT_MODECHANGED)) {
+                        EXPECT_TRUE(handler != nullptr);
+                        dsFrontPanelModeChange = handler;
+                    }
+
+                    return IARM_RESULT_SUCCESS;
+                }));
+	FrontPanelIndicatorImplMock frontPanelIndicatorMock;
+
+    device::FrontPanelIndicator indicatorList;
+
+        indicatorList.impl = &frontPanelIndicatorMock;
+    ON_CALL(frontPanelConfigImplMock, getIndicators())
+        .WillByDefault(::testing::Return(device::List<device::FrontPanelIndicator>({indicatorList})));
+    ON_CALL(frontPanelIndicatorImplMock, getName())
```

Comment:
please reformat as asked in [readme](https://github.com/rdkcentral/rdkservices/tree/sprint/22Q4/Tests#faq)